### DISCUSSION TITLE
Code comment syntax?

### DISCUSSION THREAD

**Discussion by derekmahar** [johnkerl/miller#1154] [Q&A] [answered]
Does the Miller DSL support code comments?

**Comment by derekmahar** [comment]
I learned from example [Sieve of Eratosthenes](https://miller.readthedocs.io/en/latest/programming-examples/#sieve-of-eratosthenes) in the [Miller documentation](https://miller.readthedocs.io/) that the Miller DSL supports line comments that begin with the hash character (#).  Does any other section of the Miller documentation mention this comment syntax?

**Comment by johnkerl** [comment, marked as answer]
@derekmahar this should be clearer in the docs! :)

**Reply by johnkerl** [reply]
#1165 updates:

* https://miller.readthedocs.io/en/latest/miller-programming-language/
* https://miller.readthedocs.io/en/latest/reference-dsl-syntax/

**Comment by janxkoci** [comment]
BTW one aspect of comments in miller scripts still trips me up: comments are supported in DSL (e.g. in scripts loaded with `mlr put -f`) but not in "normal" miller scripts (i.e. loaded with `mlr -s`). In regular miller scripts only the first line can have a `#` to allow shebangs, but using the `#` anywhere else leads to an error. :confused: 

Can we have comments in both types of scripts, pretty please? :grin:

**Reply by johnkerl** [reply]
https://github.com/johnkerl/miller/issues/1343
